nym-vpn-desktop/src-tauri/src/commands/db.rs

```diff
@@ -0,0 +1,138 @@
+    let entry_node_location = db
+        .get_typed::<NodeLocation>(Key::EntryNodeLocation)
+        .map_err(|_| CmdError::new(CmdErrorSource::InternalError, err_msg.clone()))?;
+    let exit_node_location = db
+        .get_typed::<NodeLocation>(Key::ExitNodeLocation)
+        .map_err(|_| CmdError::new(CmdErrorSource::InternalError, err_msg.clone()))?;
+    let vpn_mode = db
+        .get_typed::<VpnMode>(Key::VpnMode)
+        .map_err(|_| CmdError::new(CmdErrorSource::InternalError, err_msg.clone()))?;
+    let ui_theme = db
+        .get_typed::<UiTheme>(Key::UiTheme)
+        .map_err(|_| CmdError::new(CmdErrorSource::InternalError, err_msg.clone()))?;
+    let entry_location_enabled = db
+        .get_typed::<bool>(Key::EntryLocationEnabled)
+        .map_err(|_| CmdError::new(CmdErrorSource::InternalError, err_msg.clone()))?;
+    let ui_root_font_size = db
+        .get_typed::<u32>(Key::UiRootFontSize)
+        .map_err(|_| CmdError::new(CmdErrorSource::InternalError, err_msg.clone()))?;
+    let monitoring = db
+        .get_typed::<bool>(Key::Monitoring)
+        .map_err(|_| CmdError::new(CmdErrorSource::InternalError, err_msg.clone()))?;
+    let autoconnect = db
+        .get_typed::<bool>(Key::Autoconnect)
+        .map_err(|_| CmdError::new(CmdErrorSource::InternalError, err_msg.clone()))?;
+
+    trace!(
+        r"
+entry_node_location: {:?}
+exit_node_location: {:?}
+vpn_mode: {:?}
+ui_theme: {:?}
+entry_location_enabled: {:?}
+ui_root_font_size: {:?}
+monitoring: {:?}
+autoconnect: {:?}",
+        entry_node_location,
+        exit_node_location,
+        vpn_mode,
+        ui_theme,
+        entry_location_enabled,
+        ui_root_font_size,
+        monitoring,
+        autoconnect
+    );
+
+    Ok(AppData {
+        entry_node_location,
+        exit_node_location,
+        vpn_mode,
+        ui_theme,
+        entry_location_enabled,
+        ui_root_font_size,
+        monitoring,
+        autoconnect,
+    })
+}
+
+#[instrument(skip(db))]
+#[tauri::command]
+pub async fn db_get(db: State<'_, Db>, key: Key) -> Result<Option<JsonValue>, CmdError> {
+    debug!("db_get");
+    db.get(key).map_err(|_| {
+        CmdError::new(
+            CmdErrorSource::InternalError,
+            format!("Failed to get key [{key}]"),
+        )
+    })
+}
+
+#[instrument(skip(db))]
+#[tauri::command]
+pub async fn db_set(
+    db: State<'_, Db>,
+    key: Key,
+    value: JsonValue,
+) -> Result<Option<JsonValue>, CmdError> {
+    debug!("db_set");
+    db.insert(key, &value).map_err(|e| match e {
+        DbError::Serialize(e) => CmdError::new(
+            CmdErrorSource::CallerError,
+            format!("Failed to insert key, bad JSON input: {e}"),
+        ),
+        _ => CmdError::new(
+            CmdErrorSource::InternalError,
+            format!("Failed to insert key: {e}"),
+        ),
+    })
+}
+
+#[instrument(skip(db))]
+#[tauri::command]
+pub async fn db_flush(db: State<'_, Db>) -> Result<usize, CmdError> {
```

Comment:
I don't think flushing operation is something we should ever expose to the frontend. if anything we should just flush on every write and call it a day

Reply:
> I don't think flushing operation is something we should ever expose to the frontend

Why that? Flush is exposed because user can quit from within the app via a button "Quit", and we want to flush the db before exit basically

Reply:
> we should just flush on every write and call it a day

pushed b8f8f04 to flush after db mutations